render.js

```diff
@@ -0,0 +1,21 @@
+const Renderer = function () {
+    const renderPosts = function (posts) {
```

Comment:
In general, I would add more helper functions here to try an separate some of the different concerns. I would say at least 2 more.